src/jaegertracing/ConfigTest.cpp

```diff
@@ -93,4 +95,113 @@ TEST(Config, testZeroSamplingParam)
 
 #endif  // JAEGERTRACING_WITH_YAML_CPP
 
+
+void setEnv(const char *variable, const char *value) {
+#ifdef WIN32
+  _putenv_s(variable, value);
+#else
+  setenv(variable, value, true);
+#endif
+}
+
+TEST(Config, testFromEnv)
+{
+    std::vector<Tag> tags;
+    tags.emplace_back("hostname", std::string("foo"));
+    tags.emplace_back("my.app.version", std::string("1.2.3"));
+
+    Config config(false,
+                  samplers::Config("probabilistic",
+                                   0.7,
+                                   "http://host34:57/sampling",
+                                   0,
+                                   samplers::Config::Clock::duration()),
+                  reporters::Config(10,
+                                    std::chrono::milliseconds(100),
+                                    false,
+                                    "host35:77",
+                                    "http://host36:56568"),
+                  propagation::HeadersConfig(),
+                  baggage::RestrictionsConfig(),
+                  "test-service",
+                  tags);
+
+    ASSERT_EQ(std::string("http://host36:56568"), config.reporter().endpoint());
+    ASSERT_EQ(std::string("host35:77"), config.reporter().localAgentHostPort());
+
+    ASSERT_EQ(10, config.reporter().queueSize());
+    ASSERT_EQ(std::chrono::milliseconds(100),
+              config.reporter().bufferFlushInterval());
+    ASSERT_EQ(false, config.reporter().logSpans());
+
+    ASSERT_EQ(.7, config.sampler().param());
+    ASSERT_EQ(std::string("probabilistic"), config.sampler().type());
+    ASSERT_EQ(std::string("http://host34:57/sampling"),
+              config.sampler().samplingServerURL());
```

Comment:
I don't think asserts above are necessary, they are just testing constructor invocation. Only need to do that after `config.fromEnv()`

Reply:
ok.